src/main/groovy/edu/oregonstate/mist/sadapi/db/SadDAO.groovy

```diff
@@ -0,0 +1,86 @@
+package edu.oregonstate.mist.sadapi.db
+
+import edu.oregonstate.mist.sadapi.core.Sad
+import edu.oregonstate.mist.sadapi.SadApplicationConfiguration
+import io.dropwizard.jdbi.DBIFactory
+import io.dropwizard.lifecycle.Managed
+import io.dropwizard.setup.Environment
+import org.skife.jdbi.v2.DBI
+import org.skife.jdbi.v2.Handle
+import javax.ws.rs.core.Response
+import javax.ws.rs.WebApplicationException
+import java.sql.Connection
+import java.sql.CallableStatement
+import java.sql.ResultSet
+import oracle.jdbc.OracleTypes
+
+class SadDAO implements Managed {
+    private SadApplicationConfiguration configuration
+    private Environment environment
+    private Handle handle
+    private Connection connection
+
+    public SadDAO(SadApplicationConfiguration configuration, Environment environment) {
+        this.configuration = configuration
+        this.environment = environment
+    }
+
+    public void start() {
+        DBIFactory factory = new DBIFactory()
+        DBI jdbi = factory.build(environment, configuration.getDatabase(), 'oracle')
+        handle = jdbi.open()
+        connection = handle.getConnection()
+    }
+
+    public List<Sad> queryAll(Long pidm) {
+        String query = '''BEGIN
+                            ? := SB_APPLICATION_DECISION.f_query_all(?);
+                          END;'''
+        CallableStatement statement = connection.prepareCall(query)
+        statement.registerOutParameter(1, OracleTypes.CURSOR)
+        statement.setLong(2, pidm)
+        statement.execute()
+        ResultSet result = (ResultSet)statement.getObject(1)
+        List<Sad> list = new ArrayList<Sad>()
+        while (result.next())
+            list.add(map(result))
+        if (list.isEmpty())
```

Comment:
Leave condition as is,.  Groovy truth is overrated, and sometimes unclear.  I recommend wrapping all these functions in try..catch blocks.   You may reconsider why you are throwing a web exception for a database error.  Secondly, only throw exceptions when no other possibility exists, or the error is truly critical and the application cannot proceed.  That is not a stylistic suggestion, but a performance imperative. Throwing exceptions is VERY expensive.